x/mint/test/mint_test.go

```diff
@@ -0,0 +1,148 @@
+
+// TestTotalSupplyIncreasesOverTime tests that the total supply of tokens
+// increases over time as new blocks are added to the chain.
+func (s *IntegrationTestSuite) TestTotalSupplyIncreasesOverTime() {
+	require := s.Require()
+
+	initialHeight := int64(1)
+	laterHeight := int64(20)
+
+	err := s.cctx.WaitForNextBlock()
+	require.NoError(err)
+
+	initalSupply := s.getTotalSupply(initialHeight)
+
+	_, err = s.cctx.WaitForHeight(laterHeight)
+	require.NoError(err)
+	laterSupply := s.getTotalSupply(laterHeight)
+
+	require.True(initalSupply.AmountOf(app.BondDenom).LT(laterSupply.AmountOf(app.BondDenom)))
+}
+
+// TestInflationRate verifies that the inflation rate each year matches the
+// expected rate of inflation. See the README.md for the expected rate of
+// inflation.
+func (s *IntegrationTestSuite) TestInflationRate() {
+	require := s.Require()
+
+	type testCase struct {
+		year int
+		want sdktypes.Dec
+	}
+	testCases := []testCase{
+		{year: 0, want: sdktypes.MustNewDecFromStr("8.00")},
+		{year: 1, want: sdktypes.MustNewDecFromStr("7.20")},
+		{year: 2, want: sdktypes.MustNewDecFromStr("6.48")},
+		{year: 3, want: sdktypes.MustNewDecFromStr("5.832")},
+		{year: 4, want: sdktypes.MustNewDecFromStr("5.2488")},
+		{year: 5, want: sdktypes.MustNewDecFromStr("4.72392")},
+		// Note: since the testnode takes time to create blocks, test cases
+		// for years 6+ will time out give the current TimeIotaMs.
+	}
+
+	genesisTime, err := s.cctx.GenesisTime()
+	require.NoError(err)
+
+	lastYear := testCases[len(testCases)-1].year
+	lastTimestamp := genesisTime.Add(time.Duration((lastYear + 1) * minttypes.NanosecondsPerYear))
+	_, err = s.cctx.WaitForTimestampWithTimeout(lastTimestamp, 20*time.Second)
+	require.NoError(err)
+
+	for _, tc := range testCases {
+		startTimestamp := genesisTime.Add(time.Duration(tc.year * minttypes.NanosecondsPerYear))
+		endTimestamp := genesisTime.Add(time.Duration((tc.year + 1) * minttypes.NanosecondsPerYear))
+
+		startHeight, err := s.cctx.HeightForTimestamp(startTimestamp)
+		require.NoError(err)
+		endHeight, err := s.cctx.HeightForTimestamp(endTimestamp)
+		require.NoError(err)
+
+		inflationRate := s.estimateInflationRate(startHeight, endHeight)
+		actualError := inflationRate.Sub(tc.want).Abs()
+		marginOfError := sdktypes.MustNewDecFromStr("0.01")
+		if marginOfError.GT(actualError) {
+			s.Failf("TestInflationRate failure", "inflation rate for year %v is %v, want %v with a .01 margin of error", tc.year, inflationRate, tc.want)
+		}
+	}
+}
+
+func (s *IntegrationTestSuite) getTotalSupply(height int64) sdktypes.Coins {
+	require := s.Require()
+
+	options := &client.ABCIQueryOptions{Height: height}
+	result, err := s.cctx.Client.ABCIQueryWithOptions(
+		context.Background(),
+		"/cosmos.bank.v1beta1.Query/TotalSupply",
+		nil,
+		*options,
+	)
+	require.NoError(err)
+
+	var txResp banktypes.QueryTotalSupplyResponse
+	require.NoError(proto.Unmarshal(result.Response.Value, &txResp))
+	return txResp.GetSupply()
+}
```

Comment:
[not blocking]
this works for me locally, wdyt?
```suggestion
func (s *IntegrationTestSuite) getTotalSupply(height int64) sdktypes.Coins {
	require := s.Require()

	bqc := banktypes.NewQueryClient(s.cctx.GRPCClient)
	ctx := metadata.AppendToOutgoingContext(context.Background(), grpctypes.GRPCBlockHeightHeader, fmt.Sprintf("%d", height))

	resp, err := bqc.TotalSupply(ctx, &banktypes.QueryTotalSupplyRequest{})
	require.NoError(err)

	return resp.Supply
}
```